docs/advanced-usage.rst

```diff
@@ -448,3 +448,39 @@ To enable this simply define environment variable `SSLKEYLOGFILE`:
 
 Then configure the key logfile in `Wireshark <https://wireshark.org>`_, see
 `Wireshark TLS Decryption <https://wiki.wireshark.org/TLS#TLS_Decryption>`_ for instructions.
+
+Custom SSL Contexts
+-------------------
+
+.. versionadded:: 1.17.0
+
+It is possible to provide a specific `SSL Context objects`_ directly to urllib3
+to exercise fine-grained control over the SSL configuration used in urllib3.
+
+For the purposes of compatibility, if you intend to customise an ``SSLContext``
+object we *strongly* recommend you obtain one from the following factory
+function:
+
+.. autofunction:: urllib3.util.ssl_.create_urllib3_context
+
+Once you have a context object, you can mutate that to achieve whatever affect
+you'd like. For example, the code below loads the default SSL certificates on
+Linux systems, sets the `OP_NO_RENEGOTIATION <https://docs.python.org/3/library/ssl.html#ssl.OP_NO_RENEGOTIATION>`_  flag that isn't set by default,
+and then makes a HTTPS request to Google:
+
+.. code-block:: python
+
+    import ssl
+    from urllib3 import PoolManager
+    from urllib3.util.ssl_ import create_urllib3_context
+
+    ctx = create_urllib3_context()
+    ctx.load_default_certs()
```

Comment:
This line is already called by `create_urllib3_context`, please remove it.

Reply:
this is not and without the snippet fails: 
```    
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 696, in urlopen
    httplib_response = self._make_request(
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 397, in _make_request
    self._validate_conn(conn)
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1009, in _validate_conn
    conn.connect()
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/connection.py", line 496, in connect
    self.sock = ssl_wrap_socket(
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 416, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 460, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "/home/lotso/.asdf/installs/python/3.9.5/lib/python3.9/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/home/lotso/.asdf/installs/python/3.9.5/lib/python3.9/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/home/lotso/.asdf/installs/python/3.9.5/lib/python3.9/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<input>", line 10, in <module>
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/poolmanager.py", line 423, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 786, in urlopen
    return self.urlopen(
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 786, in urlopen
    return self.urlopen(
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 786, in urlopen
    return self.urlopen(
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    retries = retries.increment(
  File "/home/lotso/PycharmProjects/urllib3/.nox/test-3-9/lib/python3.9/site-packages/urllib3/util/retry.py", line 495, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))  # type: ignore
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='www.google.com', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)')))

```